components/wrap/WrapModal.tsx

```diff
@@ -0,0 +1,140 @@
+import { ethxABI, NETWORK_ID, PAYMENT_TOKEN_ADDRESS } from "../../lib/constants";
+import { getETHBalance, getETHxBalance } from "../../lib/getBalance";
+
+function WrapMpdal({ account, signer, show, handleClose }: any) {
+  const [ETHBalance, setETHBalance] = React.useState<string | undefined>();
+  const [ETHxBalance, setETHxBalance] = React.useState<string | undefined>();
+  const [isWrapping, setIsWrapping] = React.useState<boolean>(false);
+  const [isBalanceInsufficient, setIsBalanceInsufficient] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    (async () => {
+      const provider = new ethers.providers.InfuraProvider(
+        NETWORK_ID,
+        process.env.NEXT_PUBLIC_INFURA_ID
+      );
+
+      const ethBalance = await getETHBalance(provider, account);
+
+      const ethxBalance = await getETHxBalance(provider, account);
+
+      if (isMounted) {
+        setETHBalance(ethBalance);
+        setETHxBalance(ethxBalance);
+      }
+    })();
+
+    return () => {
+      isMounted = false;
+    };
+  }, []);
+
+  const wrapETH = async (amount: string) => {
+    const provider = new ethers.providers.InfuraProvider(
+      NETWORK_ID,
+      process.env.NEXT_PUBLIC_INFURA_ID
+    );
+      
+    const ETHx = new ethers.Contract(PAYMENT_TOKEN_ADDRESS, ethxABI, provider);
+
+    const reciept = await ETHx.connect(signer).upgradeByETH({
+      value: ethers.utils.parseEther(amount)
+    });
+
+    setIsWrapping(true);
+
+    try {
+      await reciept.wait();
+      // Wrap ETHx successfully!
+      const ethBalance = await getETHBalance(provider, account);
+      const ethxBalance = await getETHxBalance(provider, account);
+      // Update balances
+      setETHBalance(ethBalance);
+      setETHxBalance(ethxBalance);
+    } catch (error) {
+      console.error(error);
+    };
+
+    setIsWrapping(false);
+  }
+
+  const onSubmit = (e: any) => {
+    e.preventDefault();
+    const amount = Number((e.target[0].value));
+
+    if (amount > Number(ETHBalance)) {
+      setIsBalanceInsufficient(true);
+    } else {
+      setIsBalanceInsufficient(false);
+
+      if (!Number.isNaN(amount) && amount > 0) {
+        console.log("amount is valid", amount);
+        (async () => {
+          await wrapETH(amount.toString());
+        })();
+      }
+    }
+  }
+
+  return (
+    <Modal
+      show={show}
+      onHide={handleClose}
+      centered
+    >
+      <Modal.Header>
+        <Modal.Title>Wrap ETH for Streaming</Modal.Title>
+        <button
+          type="button"
+          className="close"
+          data-dismiss="modal"
+          aria-label="Close"
+          onClick={handleClose}
+        >
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </Modal.Header>
+      <Modal.Body>
+        <p>Current Balances</p>
+        <div style={{ padding: "0 16px" }}>
+          <p>ETH: {ETHBalance ?? '---'}</p>
+          <p>ETHx: {ETHxBalance ?? '---'}</p>
```

Comment:
The ETHx balance can always be changing in realtime. This should use `FlowingBalance` like the `Profile`.

Reply:
I don't think so, you will see that when wrapped some ETHx, there is no balance change on profile (top right corner).

and currently, when wrapped some ETHx, the balance will refetch so it will show an updated balance, which is expected.

Reply:
Not all wallets will have a balance changing. Here is an example of one that does:
https://user-images.githubusercontent.com/6885592/166179878-06acb341-5fe9-42e8-932a-627eadebeb19.mov

The wrapping works great. However, the profile balance at the top doesn't update for me. I will file a separate bug to fix that.

Reply:
I know the `sfApi.useGetRealtimeBalanceQuery` will work for streaming, but not sure for the `upgrade` event.

Reply:
I will figure out if this method works well with `useEffect`.